web_src/js/utils/color.js

```diff
@@ -0,0 +1,54 @@
+// Check similar implementation in modules/util/color.go and keep synchronization
+// Return R, G, B values defined in reletive luminance
+function getLuminanceRGB(channel) {
+  const sRGB = channel / 255;
+  const res = (sRGB <= 0.03928) ? sRGB / 12.92 : ((sRGB + 0.055) / 1.055) ** 2.4;
+  return res;
+}
+
+// Reference from: https://www.w3.org/WAI/GL/wiki/Relative_luminance
+function getLuminance(r, g, b) {
+  const R = getLuminanceRGB(r);
+  const G = getLuminanceRGB(g);
+  const B = getLuminanceRGB(b);
+  const luminance = 0.2126 * R + 0.7152 * G + 0.0722 * B;
+  return luminance;
+}
+
+const getChunksFromString = (st, chunkSize) => st.match(new RegExp(`.{${chunkSize}}`, 'g'));
+
+const convertHexUnitTo256 = (hexStr) => parseInt(hexStr.repeat(2 / hexStr.length), 16);
```

Comment:
These two functions look like magic to me .... 😁 and ... not inefficient

Reply:
This is actually with reference to [Any Hex Form Modular Approach](https://stackoverflow.com/questions/21646738/convert-hex-to-rgba) in this stackoverflow lol. But for backend, I am not sure how to implement the similar functions..

Reply:
tried to improve regexp in https://github.com/go-gitea/gitea/pull/24586/commits/520931a0bc05be23929e6738bc5a6bf0469ae1b5

Reply:
I can understand how it works, personally if I could do something by basic string operation, I won't touch regexp ....